nwapp/utils/notifier.js

```diff
@@ -28,14 +36,85 @@ function playNotificationSound() {
   audio = null;
 }
 
+
+// Cache lookup
+var urlFilePathMap = new Map();
+function getFileAtUrl(url) {
+  return new Promise(function(resolve, reject) {
+    var cachedFilePath = urlFilePathMap.get(url);
+    if(cachedFilePath) {
+      resolve(cachedFilePath);
+    }
+    else {
+      setTimeout(function() {
+        reject(new Error('Took to long to request file', url));
+      }, 750);
+
+      var tempFileStream = temp.createWriteStream();
```

Comment:
Fun fact: windows doesn't clean up its temp directory. OS X and Linux does, but only on restart.

I recommend storing them in a specific directory in tmp that you can easily clean up after yourself.